spec/models/manageiq/providers/kubernetes/container_manager/metrics_capture_spec.rb

```diff
@@ -1,78 +1,75 @@
 describe ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture do
-  before do
-    # @miq_server is required for worker_settings to work
-    @miq_server = EvmSpecHelper.local_miq_server(:is_master => true)
-    @ems_kubernetes = FactoryBot.create(
-      :ems_kubernetes,
-      :connection_configurations => [{:endpoint       => {:role => :prometheus},
-                                      :authentication => {:role => :prometheus}}],
-    ).tap { |ems| ems.authentications.each { |auth| auth.update!(:status => "Valid") } }
-    @container_project = FactoryBot.create(:container_project, :ext_management_system => @ems_kubernetes)
-
-    @node = FactoryBot.create(
-      :kubernetes_node,
-      :name                  => 'node',
-      :ext_management_system => @ems_kubernetes,
-      :ems_ref               => 'target'
-    )
-
-    @node.computer_system.hardware = FactoryBot.create(
-      :hardware,
-      :cpu_total_cores => 2,
-      :memory_mb       => 2048
-    )
-
-    @group = FactoryBot.create(
-      :container_group,
-      :ext_management_system => @ems_kubernetes,
-      :container_node        => @node,
-      :ems_ref               => 'group'
-    )
-
-    @container = FactoryBot.create(
-      :kubernetes_container,
-      :name                  => 'container',
-      :container_group       => @group,
-      :container_project     => @container_project,
-      :ext_management_system => @ems_kubernetes,
-      :ems_ref               => 'target'
-    )
+  let(:ems)               { FactoryBot.create(:ems_kubernetes_with_zone, :with_metrics_endpoint) }
+  let(:container_project) { FactoryBot.create(:container_project,          :ext_management_system => ems) }
+  let!(:group)            { FactoryBot.create(:kubernetes_container_group, :ext_management_system => ems, :container_node => node) }
+  let!(:container)        { FactoryBot.create(:kubernetes_container,       :ext_management_system => ems, :container_group => group, :container_project => container_project) }
+  let(:node) do
+    FactoryBot.create(:kubernetes_node, :name => 'node', :ext_management_system => ems, :ems_ref => 'target').tap do |node|
+      node.computer_system.hardware = FactoryBot.create(:hardware, :cpu_total_cores => 2, :memory_mb => 2_048)
+    end
```

Comment:
lets keep. But in the future if we merge https://github.com/ManageIQ/manageiq/pull/22565 this could be:

```suggestion
    FactoryBot.create(:kubernetes_node, :name => 'node', :ext_management_system => ems, :ems_ref => 'target', :hardware => FactoryBot.create(:hardware, :cpu_total_cores => 2, :memory_mb => 2_048))
```

Reply:
:+1: yeah I wanted to keep this dependent-less for simpler backport but that's a much better way to do it

Reply:
agreed. keep. reduce dependencies